### ISSUE TITLE
STM32F401 bootloader

### ISSUE THREAD

**Issue by sandmix** [stevstrong/STM32-CDC-Bootloader#3]
Is this bootloader available for 401 series ? Thanks

**Comment by stevstrong** [comment]
Unfortunatelly coulöd´t finish it yet.
But any support is welcome :).

**Comment by sandmix** [comment]
I'm still learning it, for now my skills wouldn't be enought, wanted to know what's the benefit of this bootloader compared to STM32_HID_Bootloader

**Comment by stevstrong** [comment]
Well the main difference is the underlying protocol, which makes it a bit more universal.
But the upload speed is approximately same.

**Comment by sandmix** [comment]
I was thinking it uses almost the same protocol and it detects also as HID. Maybe you have something for 401 so I can start from there or there is nothing for it ?

**Comment by stevstrong** [comment]
I was planning to program a simplified USB driver for F4 family as the existing one is not very efficient.
But I do not have anything in usable form, it is all in development phase without any description yet.

**Comment by sandmix** [comment]
Simplified you mean HID ? I star analizing info from here https://github.com/Serasidis/STM32_HID_Bootloader/commit/7a238208c86c5f4ddff0651f9b555c27e786478f 
maybe it will help.

**Comment by stevstrong** [comment]
No, simplified I mean almost bare metal, the HAL drivers are pretty bloated and it is hard to understand the concept behind.
But you could try to use it.

**Comment by sandmix** [comment]
if you have some code for F4 I can try to look into it, already checked hid bootloader and weact one. could make weact working with hid-flash changed vid and pid but it stops:
> [1209:BEBA] device is found !
> Sending <reset pages> command...
> Flashing firmware...
. 1024 Bytes
